spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/TraceWebClientBeanPostProcessor.java

```diff
@@ -395,7 +395,17 @@ public String requestHeader(ClientRequest request, String name) {
 
 		@Override
 		public Integer statusCode(ClientResponse response) {
-			return response.statusCode().value();
+			return response.rawStatusCode();
```

Comment:
something like this otherwise you still end up with the exception :P

```
  @Override public Integer statusCode(HttpResponse response) {
    int result = statusCodeAsInt(response);
    return result != 0 ? result : null;
  }
```

Reply:
Ok but `statusCode` is called only via the `statusCodeAsInt` in Brave